packages/server/src/api/controllers/row/utils.ts

```diff
@@ -45,13 +45,20 @@ export async function findRow(ctx: UserCtx, tableId: string, rowId: string) {
 }
 
 export function getTableId(ctx: Ctx) {
-  if (ctx.request.body && ctx.request.body.tableId) {
+  // top priority, use the URL first
+  if (ctx.params?.sourceId) {
+    return ctx.params.sourceId
+  }
+  // check body for a table ID
+  if (ctx.request.body?.tableId) {
     return ctx.request.body.tableId
   }
-  if (ctx.params && ctx.params.tableId) {
+  // now check for old way of specifying table ID
+  if (ctx.params?.tableId) {
```

Comment:
Should path not be a priority as well?

Reply:
true, I suppose should be sourceId then tableId path